### ISSUE TITLE
badUnnamedDerivedSignal error incorrectly triggered for jsx function blocks

### ISSUE THREAD

**Issue by russelldavis** [solidjs-community/eslint-plugin-solid#63]
**Describe the bug**
See title

**To Reproduce**
```js
const TestComponent = () => {
	const [foo, setFoo] = createSignal(0);
	return <div>{() => {
        console.log("hello");
        return foo();
    }}</div>;
};
```

**Expected behavior**
No error should be generated, because solid runs these functions in a tracked scope.

**Environment (please complete the following information):**
- OS: MacOS 12.6.1
- Node version (`node --version`): v16.18.0
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): eslint-plugin-solid@0.9.1
- `eslint` version (`npm list eslint`/`yarn why eslint`): eslint@8.30.0

**Additional context**

- [x] I would be willing to contribute a PR to fix this issue
